odoo/service/server.py

```diff
@@ -501,6 +501,21 @@ def stop(self):
         if self.httpd:
             self.httpd.shutdown()
 
+        # The longpolling/poll http route waits for maximum bus.TIMEOUT seconds
+        # for new messages in the queue. It uses a threading.Event as sync
+        # primitive with the longpolling bus thread. As no new message can
+        # arrive in the bus, all the http workers that are serving that route
+        # are stuck until the wait() timeouts (default: 50 seconds). We set
+        # the event in order to wake them all up so the http worker can fulfil
+        # the request and exit.
+        try:
+            import odoo.addons.bus.models.bus as bus
+            for events in bus.dispatch.channels.values():
```

Comment:
This breaks modularity.  We better have a simple container in `service` where modules can subscribe "resources" to free or shutdown before quitting, and `bus` would simply be one of them...